File: course/src/main/scala/example/Factory/Simple/Java/SimpleFactory.java
package example.Factory.Simple.Java;

//定义产品接口
interface Product{
	void show();
}
//以下实现了具体产品类
class Mouse implements Product {
	@Override
	public void show() {
		System.out.println("A mouse has been built");
	}
}
class Keyboard implements Product {
	@Override
	public void show(){
		System.out.println("A keyboard has been built");
	}
}
class initFactory {
	public Product produce(String name) {
		switch (name) {
			case "Mouse":
				return new Mouse();
			case "Keyboard":
				return new Keyboard();
			default:
				throw new IllegalArgumentException();
		}
	}
}
//简单使用
public class SimpleFactory {
	public static void main(String[] args) {
		initFactory simpleFactory = new initFactory();
		Product mouse = simpleFactory.produce("Mouse");
		mouse.show();
	}
}

File: course/src/main/Factory/Simple/Scala/SimpleFactory.scala
trait Product {
	def show()
}
case class Mouse() extends Product {
	def show = println("A mouse has been built")
}
case class Keyboard() extends Product {
	def show = println("A keyboard has been built")
}
object SimpleFactory  {//object代替class
def produce(name: String): Product =  name match {
	case "Mouse" =>   Mouse()
	case "Keyboard" =>   Keyboard()
}
}
object Test extends App {
	val mouse: Product = SimpleFactory.produce("Mouse")
	mouse.show()
}